chsdi/lib/helpers.py

```diff
@@ -460,15 +495,13 @@ def float_raise_nan(val):
 
 def parse_box2d(stringBox2D):
     extent = stringBox2D.replace('BOX(', '').replace(')', '').replace(',', ' ')
-    # Python2/3
-    box = map(float, extent.split(' '))
-    if not isinstance(box, list):
-        box = list(box)
+    # TODO: clean-up when only Python 3.x and no longer 2.x is in use
+    box = list(map(float, extent.split(' ')))
     return box
 
 
 def is_box2d(box2D):
-    # Python2/3
+    # TODO: clean-up when only Python 3.x and no longer 2.x is in use
```

Comment:
Looking at the code, the `if not isinstance(box2D, list)` don't make sense at all here. Moreover having a method called `is_` which return the expected value instead of `True/False` is not familiar and error prone, I mean, if you ask `is this a bbox 2D ?` you expect `yes` or `no` as answer and not the `bbox 2D`. 
So here if we already touch this code, I'll remove the `TODO comment` and remove the `if not isinstance()` which is unnecessary as every caller pass a list except for one call in https://github.com/geoadmin/mf-chsdi3/blob/fb968a4668835cc2eb90211e2683990b1ba20082/chsdi/views/search.py#L363 where the `self.bbox` is only set here https://github.com/geoadmin/mf-chsdi3/blob/fb968a4668835cc2eb90211e2683990b1ba20082/chsdi/views/search.py#L36 and where it should be a string, so in this case it will anyway raise an error. I guess this code is never reached. Anyway this code will be removed in Frankfurt as the search service has been migrated in service-search-wsgi.
Finally I would rename the function to `validate_box2d()`:
```python
def validate_box2d(box2D):
    # Bottom left to top right only
    if len(box2D) != 4 or box2D[0] > box2D[2] or box2D[1] > box2D[3]:
        raise ValueError('Invalid box2D.')
    return box2D
```

Reply:
Yes, there are some examples where there were already some `list()` conversions in the code, that might not be needed. But I rather left them, as they don't cause any harm, rather than probably overlooking some call, for which the conversion would make sense.
I could do the changes now, but this would take some more time and be slightly risky.
So I would suggest, to do this in context of post-migration clean-up, when switching to Python3 completely.

Reply:
Sorry I don't see any risk here, the code is pretty clear and simple, and if you don't do it now within this PR the first analysis done here will be very likely forgotten.

Reply:
Done